xds/src/test/java/io/grpc/xds/ControlPlaneRule.java

```diff
@@ -144,6 +147,38 @@ public Server getServer() {
     NameResolverRegistry.getDefaultRegistry().deregister(nameResolverProvider);
   }
 
+  /**
+   * Will shutdown existing server if needed.
+   * Then creates a new server in the same way as {@link #starting(Description)} and starts it.
+   */
+  public void restartTdServer() {
+
+    if (getServer() != null && !getServer().isShutdown()) {
+      getServer().shutdownNow();
+      try {
+        if (!getServer().awaitTermination(5, TimeUnit.SECONDS)) {
+          logger.log(Level.SEVERE, "Timed out waiting for server shutdown");
+        }
+      } catch (InterruptedException e) {
+        throw new AssertionError("unable to shut down control plane server", e);
+      }
+    }
+
+    try {
+      createAndStartTdServer();
+    } catch (Exception e) {
+      throw new AssertionError("unable to restart the control plane server", e);
+    }
+  }
+
+  private void createAndStartTdServer() throws IOException {
```

Comment:
What does "Td" stand for?

Reply:
Transdermal

I've replaced it with Xds which seems more obvious.